### DISCUSSION TITLE
Bookmark code search queries

### DISCUSSION THREAD

**Discussion by garthhumphreys** [community/community#38742] [Other Feature Feedback, Questions, & Ideas]
### Select Topic Area

Product Feedback

### Body

Hi, 

First of all great job to the Team on this new Feature, it's really fantastic!

I only have one feature request, **Bookmarks**!

I would like to save my search queries and even possibly manage them via folders if possible. Additionally, having a way to just view previous searches, i.e. Code Search History, would be great too.

In this way, I can either review what it was I was searching last or even manage bookmarked searches.

Overall, I think it's great work and look forward to seeing it rolled out to everyone!

**Comment by ossanna16** [comment]
Hi there @garthhumphreys and welcome to our community! Thank you for the feature request!

**Comment by ndonewar** [comment]
Hi @garthhumphreys, thanks for the suggestions!

The new code search does support saved searches, via the "Save search" button (near the top right on a results page). The new code search should also show recent searches. Both of these will display below the search input field, either when the field is empty or when the text in the field matches a search name or recent query. We're also looking into ways to improve the management of saved searches.

<img width="1199" alt="image" src="https://user-images.githubusercontent.com/310986/201753970-b94ff974-9e36-490b-875c-0b97c24c31cb.png">

**Reply by garthhumphreys** [reply]
Fantastic! I'll give it a try. Thanks.

**Comment by DanduboinaRajasekhar** [comment]
Books store on the cloud

**Comment by DanduboinaRajasekhar** [comment]
Books store on the cloud
